Source/Tools/Mangos.Extractor/Functions.cs

```diff
@@ -233,18 +234,35 @@ public enum Types
 
         public static void ExtractUpdateFields()
         {
+            var TBC = 0;
+            var alpha = 0;
+            var versInfo = FileVersionInfo.GetVersionInfo("Wow.exe");
             var f = new FileStream("wow.exe", FileMode.Open, FileAccess.Read, FileShare.Read, 10000000);
             var r1 = new BinaryReader(f);
             var r2 = new StreamReader(f);
-            var o = new FileStream("Global.UpdateFields.cs", FileMode.Create, FileAccess.Write, FileShare.None, 1024);
+            var o = new FileStream(versInfo.FileMajorPart + "." + versInfo.FileMinorPart + "."+ versInfo.FileBuildPart + "." + versInfo.FilePrivatePart + "_Global.UpdateFields.cs", FileMode.Create, FileAccess.Write, FileShare.None, 1024);
             var w = new StreamWriter(o);
             int FIELD_NAME_OFFSET = SearchInFile(f, "CORPSE_FIELD_PAD");
             int OBJECT_FIELD_GUID = SearchInFile(f, "OBJECT_FIELD_GUID") + 0x400000;
             int FIELD_TYPE_OFFSET = SearchInFile(f, OBJECT_FIELD_GUID);
-            if (FIELD_NAME_OFFSET == -1)
-            {
-                FIELD_NAME_OFFSET = SearchInFile(f, "CORPSE_FIELD_FLAGS");
+            #if DEBUG
+            MessageBox.Show("FIELD_NAME_OFFSET " + FIELD_NAME_OFFSET + " OBJECT_FIELD_GUID " + OBJECT_FIELD_GUID + " FIELD_TYPE_OFFSET " + FIELD_TYPE_OFFSET);
+            #endif
+            if (FIELD_NAME_OFFSET == -1) // pre 1.5 vanilla support
+            {
+                FIELD_NAME_OFFSET = SearchInFile(f, "CORPSE_FIELD_FLAGS");
             }
+            if (FIELD_NAME_OFFSET == -1) // alpha support
```

Comment:
@Daribon  - Won't this if (which is the same as the one above) cause the FIELD_NAME_OFFSET from line 253 to be replaced by line 257 and again in line 263 ?

Perhaps FIELD_NAME_OFFSET needs to be checked as well ?

Reply:
FIELD_TYPE_OFFSET is only changed in TBC, it's the same through alpha and entire vanilla.
However FIELD_NAME_OFFSET is changed twice in vanilla, once in patch 0.6 and then another time in patch 1.5.

It first checks for FIELD_NAME_OFFSET for 1.5+, if it cannot be found it checks for pre 1.5, if that cannot be found it checks for FIELD_NAME_OFFSET from pre 0.6.